### PR TITLE
PHOENIX-5348: Fix flaky test: testIndexRebuildTask

### PR DESCRIPTION
**Pull request by gokceni**

### CODE REVIEW

**Comment by gokceni** [comment]
@lhofhansl @ChinmaySKulkarni

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DropTableWithViewsIT.java [hunk 1]
```diff
@@ -166,9 +168,14 @@ public static void assertTaskColumns(Connection conn, String expectedStatus, PTa
...
 
+        if (expectedTableName != null) {
+            String tableName = rs.getString(PhoenixDatabaseMetaData.TABLE_NAME);
+            assertEquals(expectedTableName, tableName);
+        }
+
         if (expectedData != null) {
             String data = rs.getString(PhoenixDatabaseMetaData.TASK_DATA);
-            assertEquals(expectedData, data);
+            assertTrue(data.contains(expectedData));
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DropTableWithViewsIT.java]: I ran `DropTableWithViewsIT` and it looks like `data` just gets the value `{"TaskDetails":"COMPLETED"}`, whereas when running `IndexMetadataIT`, it gets a json such as `{"IndexName":"R_ASYNCIND_N000002","RebuildAll":true,"JobID":"job_local1895022878_0001","TaskDetails":"COMPLETED"}`. 

To make this check more accurate, can we pass in a key and expected value and parse the data accordingly? Just checking `contains` may cause us to miss actual failures or even lead to false alarms if we change values stored in the `TASK_DATA`.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 2]
```diff
@@ -129,43 +128,48 @@ public void testIndexRebuildTask() throws Throwable {
...
             Admin admin = queryServices.getAdmin();
             TableName tableName = TableName.valueOf(viewIndexTableName);
             admin.disableTable(tableName);
-            admin.truncateTable(tableName, false);
+            admin.truncateTable(tableName, true);
 
             data = "{IndexName:" + indexName + ", DisableBefore:true}";
 
-            // Add a new task (update status to created) to System.Task to rebuild indexes
+            // Add a new task to System.Task to rebuild indexes
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Can you move this comment back to the line just before we `addTask`?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 3]
```diff
@@ -129,43 +128,48 @@ public void testIndexRebuildTask() throws Throwable {
...
+            conn.createStatement().execute("DELETE " + " FROM " + PhoenixDatabaseMetaData.SYSTEM_TASK_NAME
+                    + " WHERE TABLE_NAME ='" + viewName  + "'");
+            conn.commit();
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: I guess we are doing this to remove any possible tasks that may already be present in SYSTEM.TASK for the `viewName`. If so, can you please add a comment here about this?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 4]
```diff
@@ -129,43 +128,48 @@ public void testIndexRebuildTask() throws Throwable {
...
+            Thread.sleep(1);
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Instead of sleeping without checking, perhaps query SYSTEM.TASK to check that the tasks for that `viewName` have been successfully deleted, in a loop bounded by some number of retries. Then continue if all tasks have been deleted. If some tasks still remain even after said number of retries, we can perhaps fail the test since I believe this would eventually fail the test anyways due to the leftover tasks for that `viewName`, right?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 5]
```diff
@@ -129,43 +128,48 @@ public void testIndexRebuildTask() throws Throwable {
...
+            waitForTaskState(conn, PTable.TaskType.INDEX_REBUILD, viewName, PTable.TaskStatus.COMPLETED);
 
             // See that index is rebuilt and confirm index has rows
             Table htable= queryServices.getTable(Bytes.toBytes(viewIndexTableName));
             count = getUtility().countRows(htable);
             assertEquals(numOfValues, count);
         } finally {
-            conn.createStatement().execute("DELETE " + " FROM " + PhoenixDatabaseMetaData.SYSTEM_TASK_NAME);
+            conn.createStatement().execute("DELETE " + " FROM " + PhoenixDatabaseMetaData.SYSTEM_TASK_NAME
+            + " WHERE TABLE_NAME ='" + viewName  + "'");
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: This should all be inside the `if(conn != null)` block. Also, instead of doing this in a finally block, does it make sense to have an `@After` method to clean up the SYSTEM.TASK table? That way, if we add more tests here, we don't have to repeat this code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DropTableWithViewsIT.java (hunk 1 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/DropTableWithViewsIT.java]: Key is a bit complicated:
PRIMARY KEY (" + TASK_TYPE + "," + TASK_TS + " ROW_TIMESTAMP," + TENANT_ID + "," + TABLE_SCHEM + "," +
            TABLE_NAME + "))
I will try to move to key

Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/DropTableWithViewsIT.java]: I meant, key="IndexName" and value="<name to find" which we then parse from the `TASK_DATA` string, rather than just doing a `contains` within the entire string. I wasn't talking about the primary key, sorry for the confusion.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 3 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Actually, we don't need to remove this. I realized that Lars changed connection settings to autocommit=false. When I commit after setting status to Created, I don't need to delete the task.

Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: @gokceni I thought the point of this was to remove any possible leftover tasks from any previous operations, so that we ensure that all future checks only correspond to tasks created via the tests. Is this not the case? I don't understand how this is related to autocommit.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 5 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: @ChinmaySKulkarni, there is one task I create before this point in the same test function. I run an IndexRebuild task, and then I go ahead and remove the contents of index via admin api and re-run it to see if it builds. We can do it two ways: delete the task record and create a new record or just update the task record and set its status so that it gets picked up. Same thing really in terms of what we want to achieve.

**Comment by gokceni** [comment]
@dbwong the main change in the actual code is to add a timestamp parameter to queryTaskTable method so that we get the Task with the given timestamp. It is tested in the IndexRebuildTaskIT (in code/test path already). If you have a particular test suggestion, I can try looking to implement that.

File: phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java [hunk 6]
```diff
@@ -230,7 +231,13 @@ public static void deleteTask(PhoenixConnection conn, PTable.TaskType taskType,
                 taskQuery += " AND " + PhoenixDatabaseMetaData.TASK_DATA + " LIKE '%" + indexName + "%'";
             }
 
-        return populateTasks(connection, taskQuery);
+            List<TaskRecord> result = populateTasks(connection, taskQuery);
+            if (ts != null) {
+                // Adding TASK_TS to the where clause did not work. It returns empty when directly querying with the timestamp.
+                result.removeIf(p -> !p.getTimeStamp().equals(ts));
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java]: This might complain for some branches which need to be source compatible with Java 7 instead of 8

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 7]
```diff
@@ -188,7 +187,17 @@ public static void waitForTaskState(Connection conn, PTable.TaskType taskType, P
 
             String taskStatus = null;
 
-            if (rs.next()) {
+            while (rs.next()) {
```
Comment (dbwong) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: I will attempt to allude to the type of testing I'm referring to here.  I have not yet spent enough time to understand all the logic but this appears to be 2 changes.  1, the data layer may return multiple results for our query.  2, we need to test against the results and only.  However, since the logic today couples these concerns testing that our logic handles is not straight forward.  We should be able to write 3 unit tests, one that our logic handles multiple results and two we look for the expected name and the expected time stamp.  While I agree that your IT tests likely check this behavior the very act of making code unit testable suggests better boundaries.  @gokceni

File: phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java [hunk 8]
```diff
@@ -230,7 +232,20 @@ public static void deleteTask(PhoenixConnection conn, PTable.TaskType taskType,
                 taskQuery += " AND " + PhoenixDatabaseMetaData.TASK_DATA + " LIKE '%" + indexName + "%'";
             }
 
-        return populateTasks(connection, taskQuery);
+            List<TaskRecord> taskRecords = populateTasks(connection, taskQuery);
+            List<TaskRecord> result = new ArrayList<TaskRecord>();
+            if (ts != null) {
```
Comment (dbwong) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java]: Let me try again I accidentally commented in test code.  Here we have a simple logic change where we query against a property.  This is tested in the ITs but it is good to test with a small simple unit test.  However, it might pose a bit more difficult to test if there isn't a boundary between the data layer and the logic.  This may inform how to shape our code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 9]
```diff
@@ -129,45 +128,44 @@ public void testIndexRebuildTask() throws Throwable {
...
             admin.disableTable(tableName);
-            admin.truncateTable(tableName, false);
+            admin.truncateTable(tableName, true);
 
             data = "{IndexName:" + indexName + ", DisableBefore:true}";
 
-            // Add a new task (update status to created) to System.Task to rebuild indexes
+            // Add a new task to System.Task to rebuild indexes with a new timestamp.
+            //Thread.sleep(1);
+            //startTs = new Timestamp(EnvironmentEdgeManager.currentTimeMillis());
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Can we remove this commented code?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 10]
```diff
@@ -129,45 +128,42 @@ public void testIndexRebuildTask() throws Throwable {
             Task.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
                     TENANT1, null, viewName,
                     PTable.TaskStatus.CREATED.toString(), data, null, startTs, null, true);
-
-
+            conn.commit();
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Why do we need `conn.commit` here?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 11]
```diff
@@ -188,7 +185,17 @@ public static void waitForTaskState(Connection conn, PTable.TaskType taskType, P
 
             String taskStatus = null;
 
-            if (rs.next()) {
+            while (rs.next()) {
+                String tableName = rs.getString(PhoenixDatabaseMetaData.TABLE_NAME);
+                if (!tableName.equals(expectedTableName)) {
```
Comment (lhofhansl) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: I think it would be better to filter for the table name in the SELECT statement above (i.e. add it to the WHERE clause)

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 10 above)
Comment (lhofhansl) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Likewise I don't understand why we need the commit here (and below)

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 12]
```diff
@@ -83,6 +82,7 @@ private String generateDDL(String format) {
     @Test
     public void testIndexRebuildTask() throws Throwable {
         String baseTable = generateUniqueName();
+        String viewName = generateUniqueName();
         Connection conn = null;
         Connection viewConn = null;
```
Comment (swaroopak) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: nit: Could you please rename this to tenantConn, makes it more readable.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 13]
```diff
@@ -129,45 +128,42 @@ public void testIndexRebuildTask() throws Throwable {
             Task.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: @gokceni Above, we have upserted data into the view. Won't this automatically trigger an index build i.e. won't global view index get these populated rows already? Why do we need to explicitly run IndexRebuildTask?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 14]
```diff
@@ -129,45 +128,42 @@ public void testIndexRebuildTask() throws Throwable {
...
-
-
+            conn.commit();
             task.run();
+            waitForTaskState(conn, PTable.TaskType.INDEX_REBUILD, viewName, startTs, PTable.TaskStatus.COMPLETED);
 
             String viewIndexTableName = MetaDataUtil.getViewIndexPhysicalName(baseTable);
             ConnectionQueryServices queryServices = conn.unwrap(PhoenixConnection.class).getQueryServices();
             int count = getUtility().countRows(queryServices.getTable(Bytes.toBytes(viewIndexTableName)));
             assertTrue(count == numOfValues);
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Not part of your diffs, but we can replace this with `assertEquals` instead

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java [hunk 15]
```diff
@@ -129,45 +128,42 @@ public void testIndexRebuildTask() throws Throwable {
...
+            admin.truncateTable(tableName, true);
 
             data = "{IndexName:" + indexName + ", DisableBefore:true}";
 
-            // Add a new task (update status to created) to System.Task to rebuild indexes
+            // Update task to to rebuild indexes.
             Task.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
                     TENANT1, null, viewName,
                     PTable.TaskStatus.CREATED.toString(), data, null, startTs, null, true);
+            conn.commit();
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Why do we need `conn.commit` here?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 11 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: +1 to that. I think that would also not require us to have the `continue` statements in case there are no matches.

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java [hunk 16]
```diff
@@ -199,6 +200,7 @@ public void run() {
                             Task.addTask(connForTask, taskRecord.getTaskType(), taskRecord.getTenantId(), taskRecord.getSchemaName(),
                                     taskRecord.getTableName(), PTable.TaskStatus.STARTED.toString(),
                                     taskRecord.getData(), taskRecord.getPriority(), taskRecord.getTimeStamp(), null, true);
+                            connForTask.commit();
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java]: nit: null check for `connForTask`

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java [hunk 17]
```diff
@@ -255,6 +257,7 @@ public static void setEndTaskStatus(PhoenixConnection connForTask, Task.TaskReco
             Task.addTask(connForTask, taskRecord.getTaskType(), taskRecord.getTenantId(), taskRecord.getSchemaName(),
                     taskRecord.getTableName(), taskStatus, data, taskRecord.getPriority(),
                     taskRecord.getTimeStamp(), endTs, true);
+            connForTask.commit();
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java]: nit: null check for `connForTask`

File: phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java [hunk 18]
```diff
@@ -230,7 +232,20 @@ public static void deleteTask(PhoenixConnection conn, PTable.TaskType taskType,
...
+                // Adding TASK_TS to the where clause did not work. It returns empty when directly querying with the timestamp.
+                for (TaskRecord tr : taskRecords) {
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java]: @gokceni Why do we need these changes for fixing the test?

**Comment by swaroopak** [comment]
@gokceni Let me know when you update this PR.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 15 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: @gokceni I still don't understand why we are committing here.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 11 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: @gokceni can you please address these comments?

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java (hunk 16 above)
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java]: @gokceni why do we need to commit here?

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java (hunk 17 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java]: Also, this and the previous change in this file seem outside the scope of this JIRA which is just to fix the flaky test.

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java [hunk 19]
```diff
@@ -107,6 +107,7 @@
             Task.addTask(conn.unwrap(PhoenixConnection.class ), taskRecord.getTaskType(), taskRecord.getTenantId(), taskRecord.getSchemaName(),
                     taskRecord.getTableName(), PTable.TaskStatus.STARTED.toString(), jsonObject.toString(), taskRecord.getPriority(),
                     taskRecord.getTimeStamp(), null, true);
+            conn.commit();
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java]: @gokceni what are we committing here? Also, this change seems outside the scope of this JIRA again. Is there any change you made which would now require a `conn.commit` after adding a task?

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java [hunk 20]
```diff
@@ -162,7 +163,9 @@ private String getJobID(String data) {
             Cluster cluster = new Cluster(configuration);
 
             Job job = cluster.getJob(org.apache.hadoop.mapreduce.JobID.forName(jobID));
-
+            if (job == null) {
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java]: @gokceni where is this used?

File: phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java (hunk 18 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java]: ping @gokceni please address these review comments from earlier. Also, would this change affect any other tasks wherein timestamp is specified?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 10 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: Removed the commit after adding task. The connection is setAutocommit(false)

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 11 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: addressed

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 12 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: done

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 13 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: removed

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 14 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: done

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java (hunk 15 above)
Comment (gokceni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java]: removed

File: phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java (hunk 18 above)
Comment (gokceni) [reply phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java]: @ChinmaySKulkarni Your previous comments about uniquely getting a task made me question about task query. I realized that there might be a case where we have tasks for the same index and table but with a different timestamp. I change the code to handle that.

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java (hunk 16 above)
Comment (gokceni) [reply phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java]: removed

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java (hunk 19 above)
Comment (gokceni) [reply phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java]: removed

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java (hunk 20 above)
Comment (gokceni) [reply phoenix-core/src/main/java/org/apache/phoenix/coprocessor/tasks/IndexRebuildTask.java]: When you kick the task and it runs index tool, cluster.getJob doesn't return a job immediately sometimes. We have to come back and requery the job status when we find the job from cluster.getJob

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java (hunk 16 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java]: @gokceni looks like `connForTask.commit()` is still there.

**Comment by gjacoby126** [comment]
JIRA has already been committed, closing PR.
